server.js

```diff
@@ -1,33 +1,107 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import netflixData from './data/netflix-titles.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.set('view engine', 'ejs')
+app.use(express.static("public"))
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.redirect(301, '/netflix?page=1&perPage=12')
 })
 
+// display all the
+app.get('/netflix', (req, res) => {
+
+  let page = parseInt(req.query["page"], 10) ?? 0;
+  let perPage = parseInt(req.query["perPage"], 10) ?? 5;
+
+  let sliceStart = page * perPage;
+  let sliceEnd = sliceStart + perPage;
+  res.render('pages/shows', {
+    allData: netflixData.slice(sliceStart, sliceEnd),
+    endpoint: '/netflix'
+  })
+})
+
+
+app.get('/movies', (req, res) => {
+  let movies = netflixData.filter((item) => item.type === "Movie")
+
+  let page = parseInt(req.query["page"], 10) ?? 0;
+  let perPage = parseInt(req.query["perPage"], 10) ?? 5;
+
+  let sliceStart = page * perPage;
+  let sliceEnd = sliceStart + perPage;
+
+  res.render('pages/movies', {
+    movies: movies.slice(sliceStart, sliceEnd),
+    endpoint: '/movies'
+  })
+})
+
+
+
+app.get('/tv-shows', (req, res) => {
+  let tvshows = netflixData.filter((item) => item.type === "TV Show")
+
+  let page = parseInt(req.query["page"], 10) ?? 0;
+  let perPage = parseInt(req.query["perPage"], 10) ?? 5;
+
+  let sliceStart = page * perPage;
+  let sliceEnd = sliceStart + perPage;
+
+  res.render('pages/tv-shows', {
+    tvshows: tvshows.slice(sliceStart, sliceEnd),
+    endpoint: '/tv-shows'
+
+  });
+})
+
+
+app.get('/netflix/id/:id', (req, res) => {
+  const id = req.params.id
+  const show = netflixData.find((item) => item.show_id === +id)
+
+  if (!show) { res.status(404).send() }
+
+  res.render('pages/show', {
+    show: show,
+  });
+});
+
+
+
+// http://localhost:8080/api/netflix?country=nigeria&year=2018
+app.get('/api/netflix', (req, res) => {
```

Comment:
Don't forget to use RESTful routes. The route should be the pluralised version of the data you're returning. So if this is returning a 'Show', then this endpoint should be `/shows` and then using a query parameter for the type. So ending up with an endpoint like this: `/shows` - returns everything, `/shows?year=2019` - returns only shows from 2019